ctapipe/reco/HillasReconstructor.py

```diff
@@ -99,10 +98,14 @@ class that reconstructs the direction of an atmospheric shower
 
     """
 
-    def __init__(self, config=None, parent=None, **kwargs):
+    def __init__(self, event=None, config=None, parent=None, **kwargs):
         super().__init__(config=config, parent=parent, **kwargs)
         self.hillas_planes = {}
-        self.divergent_mode = False
+        self.mode = "CameraFrame"
+        self.event = event
+
+        # dictionary to store the telescope-wise image directions
+        # to be projected on the ground and corrected in case of mispointing
         self.corrected_angle_dict = {}
```

Comment:
Again, event wise data should not be stored as state on the class.

Reply:
This is done because the correction happens in `initialize_hillas_planes` and is used in `estimate_core_position`.
I guess one could move it there, but this would mean modify `estimate_core_position` to read more inputs and put things like focal length as a state.